MoreLinq/EndsWith.cs

```diff
@@ -29,21 +29,24 @@ static partial class MoreEnumerable
         /// <typeparam name="T">Type of elements.</typeparam>
         /// <param name="first">The sequence to check.</param>
         /// <param name="second">The sequence to compare to.</param>
+        /// <param name="comparer">Optional equality comparer, defaults to <see cref="EqualityComparer{T}.Default" />.</param>
         /// <returns>
         ///     <c>true</c> if <paramref name="first" /> ends with elements equivalent to <paramref name="second" />.
         /// </returns>
         /// <remarks>
         ///     This is the <see cref="IEnumerable{T}" /> equivalent of <see cref="string.EndsWith(string)" /> and
-        ///     it calls <see cref="object.Equals(object)" /> on pairs of elements at the same index.
+        ///     it calls <see cref="IEqualityComparer{T}.Equals(T,T)" /> on pairs of elements at the same index.
         /// </remarks>
-        public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second, IEqualityComparer<T> comparer = null)
         {
             if (first == null) throw new ArgumentNullException("first");
             if (second == null) throw new ArgumentNullException("second");
 
+            comparer = comparer ?? EqualityComparer<T>.Default;
+
             using (var firstIter = first.Reverse().GetEnumerator())
```

Comment:
The implementation of `EndsWith` is redundant with `StartsWith` and adds no value because anyone can call `Reverse` on both input sequences and use `StartsWith`. `EndWith` should either be removed or implemented doing the *real work* more effectively.